esx_service/vmdk_ops.py

```diff
@@ -170,55 +169,59 @@ def createVMDK(vmdk_path, vm_name, vol_name, opts={}, vm_uuid=None, tenant_uuid=
         return cloneVMDK(vm_name, vmdk_path, opts,
                          vm_uuid, datastore)
 
-    cmd = make_create_cmd(opts, vmdk_path)
-    rc, out = RunCommand(cmd)
-    if rc != 0:
-        return err("Failed to create %s. %s" % (vmdk_path, out))
+    if not kv.DISK_ALLOCATION_FORMAT in opts:
+        disk_format = kv.DEFAULT_ALLOCATION_FORMAT
+    else:
+        disk_format = kv.VALID_ALLOCATION_FORMATS[opts[kv.DISK_ALLOCATION_FORMAT]]
+
+    # VirtualDiskSpec
+    vdisk_spec = vim.VirtualDiskManager.FileBackedVirtualDiskSpec()
+    vdisk_spec.adapterType = VMDK_ADAPTER_TYPE
+    vdisk_spec.diskType = disk_format
+
+    if  kv.SIZE in opts:
+        vdisk_spec.capacityKb = convert.convert_to_KB(opts[kv.SIZE])
+    else:
+        vdisk_spec.capacityKb = convert.convert_to_KB(kv.DEFAULT_DISK_SIZE)
+
+    # Form datastore path from vmdk_path
+    volume_datastore_path = vmdk_utils.get_datastore_path(vmdk_path)
+
+    si = get_si()
+    task = si.content.virtualDiskManager.CreateVirtualDisk(
+        name=volume_datastore_path, spec=vdisk_spec)
+    try:
+        wait_for_tasks(si, [task])
+    except vim.fault.VimFault as ex:
+        return err("Failed to create volume: {0}".format(ex.msg))
+
+    # Handle vsan policy
+    if kv.VSAN_POLICY_NAME in opts:
+        if not vsan_policy.set_policy_by_name(vmdk_path, opts[kv.VSAN_POLICY_NAME]):
+            logging.error("Could not set policy: %s to volume %s",
```

Comment:
Maybe we could just drop the logging here and delete the key from opts, since we'll get the warning from vsan_info.set_policy:
```
rc, out = vmdk_ops.RunCommand(OBJTOOL_SET_POLICY.format(uuid,
                                                            policy_string))
    if rc != 0:
        logging.warning("Failed to set policy for %s : %s", vmdk_path, out)
```